cartoframes/data/utils.py

```diff
@@ -0,0 +1,150 @@
+import time
+import binascii as ba
+
+from warnings import warn
+from carto.exceptions import CartoException, CartoRateLimitException
+
+from ..columns import Column
+
+try:
+    import geopandas
+    HAS_GEOPANDAS = True
+except ImportError:
+    HAS_GEOPANDAS = False
+
+
+DEFAULT_RETRY_TIMES = 3
+
+GEOM_COLUMN_NAMES = [
+    'geometry',
+    'the_geom',
+    'wkt_geometry',
+    'wkb_geometry',
+    'geom',
+    'wkt',
+    'wkb'
+]
+
+LAT_COLUMN_NAMES = [
+    'latitude',
+    'lat'
+]
+
+LNG_COLUMN_NAMES = [
+    'longitude',
+    'lng',
+    'lon',
+    'long'
+]
+
+
+def compute_query(dataset):
+    if dataset._table_name and dataset._schema:
+        return 'SELECT * FROM "{0}"."{1}"'.format(dataset._schema, dataset._table_name)
+
+
+def compute_geodataframe(dataset):
+    if dataset._df is not None:
+        df = dataset._df.copy()
+        geom_column = _get_column(df, GEOM_COLUMN_NAMES)
+        if geom_column is not None:
+            df['geometry'] = _compute_geometry_from_geom(geom_column)
+        else:
+            lat_column = _get_column(df, LAT_COLUMN_NAMES)
+            lng_column = _get_column(df, LNG_COLUMN_NAMES)
+            if lat_column is not None and lng_column is not None:
+                df['geometry'] = _compute_geometry_from_latlng(lat_column, lng_column)
+            else:
+                raise ValueError('''No geographic data found. '''
+                                 '''If a geometry exists, change the column name ({0}) or '''
+                                 '''ensure it is a DataFrame with a valid geometry. '''
+                                 '''If there are latitude/longitude columns, rename to ({1}), ({2}).'''.format(
+                                     ', '.join(GEOM_COLUMN_NAMES),
+                                     ', '.join(LAT_COLUMN_NAMES),
+                                     ', '.join(LNG_COLUMN_NAMES)
+                                 ))
+        return geopandas.GeoDataFrame(df)
```

Comment:
It will fail without geopandas dependency

Reply:
Yep, we need to check the flag here. Thanks